app/database/items.go

```diff
@@ -1,8 +1,16 @@
 package database
 
+// WhereUserHasViewPermissionOnItems returns a subview of the items
+// on that the given user has `can_view_generated` >= `viewPermission`
+// basing on the given view.
+func (conn *DB) WhereUserHasViewPermissionOnItems(user *User, viewPermission string) *DB {
+	itemsPerms := NewDataStore(newDB(conn.db.New())).PermissionsGenerated().WithViewPermissionForUser(user, viewPermission)
+	return conn.Joins("JOIN ? AS rights ON rights.item_id = items.id", itemsPerms.SubQuery())
```

Comment:
As this query may be chained afterwards with others, wouldn't it better to keep a consistent alias? "permissions" seems more appropriate than "rights.

Reply:
It has been 'rights' already everywhere. Do you want me to rename it?

Reply:
fixed in this method